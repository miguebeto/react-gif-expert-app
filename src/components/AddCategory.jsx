import React, { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories, onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");
  const onInputCahnge = ({ target }) => {
    setInputValue(target.value);
  };
  const onSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim() <= 1) return; //evita que se inserten elementos vacios o de una letra al arreglo
    // setCategories((cat) => [...cat, inputValue]); //llama a la funcion que setea el estado en el componente padre
    onNewCategory( inputValue.trim() ); //llama la funcion que agrega el valor al arreglo en el componente padre
    setInputValue(" "); //borra los datos del input luego de agregarlos
  };

  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input
        type="text"
        placeholder="Buscar gifs"
        value={inputValue}
        onChange={onInputCahnge}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
  setCategories: PropTypes.func
}
