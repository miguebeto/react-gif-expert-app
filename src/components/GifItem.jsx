import React from "react";

export const GifItem = ({ title, url }) => {
  return (
    <div className="card">
      <img src={url} alt="images"/>
      <p>{title}</p>
    </div>
  );
};
