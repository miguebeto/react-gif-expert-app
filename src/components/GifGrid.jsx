import React from "react";
import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from 'prop-types';


export const GifGrid = ({ category }) => {

  const { images, loading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      <div className="card-grid">
        {loading ? <h2>Cargando...</h2>  : images.map((image) => (
          <GifItem key={image.id} {...image} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
};
