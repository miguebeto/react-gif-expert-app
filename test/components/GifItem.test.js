import { GifItem } from "../../src/components/GifItem";
import { render, screen } from "@testing-library/react";

describe("Pruebas en componente GifItem.jsx", () => {
  const title = "kilua";
  const url = "https://hunterxhunter.com/kilua.jpg";
  test("debe hacer match con el snapshop", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });
  test("debe mostrar la imagen con el url y el alt indicado", () => {
    render(<GifItem title={title} url={url} />);
    // expect(screen.getByRole('img').src).toBe( url );
    // expect(screen.getByRole('img').alt).toBe( title );

    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });
  test("debe mostrar el titulo en el componente", () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
    screen.debug();
  });
});
