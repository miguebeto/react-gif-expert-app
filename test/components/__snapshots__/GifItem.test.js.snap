// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente GifItem.jsx debe hacer match con el snapshop 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="kilua"
      src="https://hunterxhunter.com/kilua.jpg"
    />
    <p>
      kilua
    </p>
  </div>
</div>
`;
